executor/containerdexecutor/executor.go

```diff
@@ -39,10 +39,13 @@ func (w containerdExecutor) Exec(ctx context.Context, meta executor.Meta, root c
 		return err
 	}
 
-	hostsFile, err := oci.GetHostsFile(ctx, w.root)
+	hostsFile, clean, err := oci.GetHostsFile(ctx, w.root, meta.ExtraHosts)
 	if err != nil {
 		return err
 	}
+	if clean != nil {
+		defer clean()
+	}
```

Comment:
Minor nit: `GetHostsFile` never seems to return `nil`, in the cases where it might it instead returns a function which is a `nop`, which would allow you to simplify a bit here if you wanted.

Reply:
Left it for safety.